custom_components/spotcast/spotify/account.py

```diff
@@ -742,19 +743,23 @@ async def async_track_features(self, uri: str) -> dict:
         if uri is None or not uri.startswith("spotify:track:"):
             return {}
 
-        response = await self.hass.async_add_executor_job(
-            self.apis["private"].audio_features,
-            [uri],
-        )
-
-        return response[0] or {}
+        try:
+            response = await self.hass.async_add_executor_job(
+                self.apis["public"].audio_features,
+                [uri],
+            )
+            return response[0] or {}
+        except Exception:  # pylint: disable=W0718
```

Comment:
narrow the exception scope. Won't allow it otherwise, API errors are easy to scope, we shouldn't use broad-except in this case.